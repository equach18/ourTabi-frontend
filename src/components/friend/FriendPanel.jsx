import { useState, useContext } from "react";
import UserContext from "../../context/UserContext";
import FriendTabs from "./FriendTabs";
import FriendCard from "./FriendCard";
import FriendSearchForm from "./FriendSearchForm";

function FriendPanel() {
  const { friends, incomingRequests, sentRequests } = useContext(UserContext);

  const [activeTab, setActiveTab] = useState("all"); // "all", "requests", "sent", "search"
  return (
    <div className="h-full flex flex-col bg-white rounded-xl shadow p-4">
      {/* Top: Fixed content */}
      <div className="mb-4">
        <h2 className="text-lg font-bold text-zinc-800 mb-2">Friends</h2>
        <FriendTabs
          activeTab={activeTab}
          setActiveTab={setActiveTab}
          requestCount={incomingRequests.length}
          sentCount={sentRequests.length}
        />
      </div>

      {/* Scrollable content */}
      <div className="flex-1 overflow-y-auto space-y-3">
        {activeTab === "all" &&
          (friends.length > 0 ? (
            friends.map((friend) => (
              <FriendCard key={friend.userId} friend={friend} />
            ))
          ) : (
            <p className="text-gray-500">No friends yet.</p>
          ))}

        {activeTab === "requests" &&
          (incomingRequests.length > 0 ? (
            incomingRequests.map((friend) => (
              <FriendCard key={friend.id} friend={friend} isRequest />
            ))
          ) : (
            <p className="text-gray-500">No friend requests.</p>
          ))}

        {activeTab === "sent" &&
          (sentRequests.length > 0 ? (
            sentRequests.map((friend) => (
              <FriendCard key={friend.id} friend={friend} isSent />
            ))
          ) : (
            <p className="text-gray-500">No sent requests.</p>
          ))}

        {activeTab === "search" && <FriendSearchForm />}
      </div>
    </div>
  );
}

export default FriendPanel;
