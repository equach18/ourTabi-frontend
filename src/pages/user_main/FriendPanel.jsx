import { useState, useContext } from "react";
import UserContext from "../../context/UserContext";
import FriendsTabs from "./FriendsTabs";
import FriendCard from "./FriendCard";
import SearchFriends from "./SearchFriends";

function FriendPanel() {
  const { currentUser } = useContext(UserContext);

  // ðŸ”¥ Simulated data (Replace with API calls)
  const [friends, setFriends] = useState([
    { id: 1, username: "MoonShine", status: "online", avatar: "/avatar1.png" },
    { id: 2, username: "Wingless", status: "online", avatar: "/avatar2.png" },
    {
      id: 3,
      username: "SniperBird",
      status: "offline",
      avatar: "/avatar3.png",
    },
  ]);

  const [friendRequests, setFriendRequests] = useState([
    {
      id: 4,
      username: "DarkKnight",
      status: "pending",
      avatar: "/avatar4.png",
    },
  ]);

  const [sentRequests, setSentRequests] = useState([
    { id: 5, username: "ShadowFox", status: "sent", avatar: "/avatar5.png" },
  ]);

  const [activeTab, setActiveTab] = useState("all"); // "all", "requests", "sent", "search"

  return (
    <div className="bg-gray-100 p-6 rounded-lg shadow-md w-full md:w-1/3">
      <h2 className="text-2xl font-semibold mb-4">Friends</h2>

      {/* ðŸ”¥ Tab Selector */}
      <FriendsTabs
        activeTab={activeTab}
        setActiveTab={setActiveTab}
        requestCount={friendRequests.length}
        sentCount={sentRequests.length}
      />

      {/* ðŸ”¥ Show Content Based on Active Tab */}
      <div className="mt-4">
        {activeTab === "all" &&
          (friends.length > 0 ? (
            friends.map((friend) => (
              <FriendCard key={friend.id} friend={friend} />
            ))
          ) : (
            <p className="text-gray-500">No friends yet.</p>
          ))}

        {activeTab === "requests" &&
          (friendRequests.length > 0 ? (
            friendRequests.map((friend) => (
              <FriendCard key={friend.id} friend={friend} isRequest />
            ))
          ) : (
            <p className="text-gray-500">No friend requests.</p>
          ))}

        {activeTab === "sent" &&
          (sentRequests.length > 0 ? (
            sentRequests.map((friend) => (
              <FriendCard key={friend.id} friend={friend} isSent />
            ))
          ) : (
            <p className="text-gray-500">No sent requests.</p>
          ))}

        {activeTab === "search" && <SearchFriends />}
      </div>
    </div>
  );
}

export default FriendPanel;
