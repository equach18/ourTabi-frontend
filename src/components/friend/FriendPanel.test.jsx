import { screen, fireEvent } from "@testing-library/react";
import FriendPanel from "./FriendPanel";
import {
  renderWithUserProvider,
  testFriendsData,
} from "../../test-helpers/_testCommons";

describe("FriendPanel", () => {
  it("matches snapshot for default All Friends tab", () => {
    const { asFragment } = renderWithUserProvider(<FriendPanel />, {
      userContext: testFriendsData,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders All Friends tab by default", () => {
    renderWithUserProvider(<FriendPanel />, {
      userContext: testFriendsData,
    });
    expect(screen.getByText("@testUser1")).toBeInTheDocument();
  });

  it("shows incoming requests when Request tab is clicked", () => {
    renderWithUserProvider(<FriendPanel />, {
      userContext: testFriendsData,
    });
    fireEvent.click(screen.getByText(/Requests/i));
    expect(screen.getByText("@testUser2")).toBeInTheDocument();
  });

  it("shows sent requests when Sent tab is clicked", () => {
    renderWithUserProvider(<FriendPanel />, {
      userContext: testFriendsData,
    });
    fireEvent.click(screen.getByText(/Sent/i));
    expect(screen.getByText("@testUser3")).toBeInTheDocument();
  });

  it("shows message when no friends exist", () => {
    const emptyContext = {
      friends: [],
      incomingRequests: [],
      sentRequests: [],
    };

    renderWithUserProvider(<FriendPanel />, { userContext: emptyContext });
    expect(screen.getByText("No friends yet.")).toBeInTheDocument();
  });
});
