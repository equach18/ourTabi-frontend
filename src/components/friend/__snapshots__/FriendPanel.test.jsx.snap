// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FriendPanel > matches snapshot for default All Friends tab 1`] = `
<DocumentFragment>
  <div
    class="h-full flex flex-col bg-white rounded-xl shadow p-4"
  >
    <div
      class="mb-4"
    >
      <h2
        class="text-lg font-bold text-zinc-800 mb-2"
      >
        Friends
      </h2>
      <div
        class="flex flex-wrap gap-2 mb-4"
      >
        <button
          class="px-3 py-1 rounded-lg text-sm font-medium transition-colors
              bg-blue-500 text-white"
        >
          All Friends
        </button>
        <button
          class="px-3 py-1 rounded-lg text-sm font-medium transition-colors
              bg-gray-200 text-gray-700 hover:bg-gray-300"
        >
          Requests (1)
        </button>
        <button
          class="px-3 py-1 rounded-lg text-sm font-medium transition-colors
              bg-gray-200 text-gray-700 hover:bg-gray-300"
        >
          Sent (1)
        </button>
        <button
          class="px-3 py-1 rounded-lg text-sm font-medium transition-colors
              bg-gray-200 text-gray-700 hover:bg-gray-300"
        >
          Search
        </button>
      </div>
    </div>
    <div
      class="flex-1 overflow-y-auto space-y-3"
    >
      <div
        class="flex items-center justify-between bg-white p-4 rounded-lg shadow-sm mb-3"
      >
        <div
          class="flex items-center gap-3"
        >
          <img
            alt="testUser1's avatar"
            class="w-10 h-10 rounded-full object-cover"
            src="/src/assets/profilePic.jpg"
          />
          <div
            class="text-sm"
          >
            <p
              class="font-semibold"
            >
              Test1 User1
            </p>
            <p
              class="text-sm text-gray-500"
            >
              @testUser1
            </p>
          </div>
        </div>
        <div
          class="flex gap-2"
        >
          <button
            class="bg-red-500 hover:bg-red-600 text-white px-2 rounded text-sm"
          >
            X
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
